app/views/users/index.html.erb

```diff
@@ -1,8 +1,38 @@
 <h2>List of registered users</h2>
-<% @users.each do |user| %> 
+<% @users.each do |user| %>
+<% if current_user.id != user.id %>
 
 <div class="field-user">
+
+    <% if (current_user.friendships.where('user_id = :id OR friend_id = :id', id: user.id).where("status = (?)", true).exists? )%>
```

Comment:
- [ ] This view file is bloated with functionality that doesn't belong to it. 
`(current_user.friendships.where('user_id = :id OR friend_id = :id', id: user.id).where("status = (?)", true).exists? )` belongs to model. So we can create accepted_friendship scope whose status is true and make it available in the controller so it will be availble in the views. Views aren't supposed to hit the database to get data, it just displays the data provided by its corresponding controller.

Reply:
solved